lib/vspc/handlers.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vspc
+
+import (
+	"bytes"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/vic/pkg/telnet"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+// handler is the handler struct for the vspc
+type handler struct {
+	vspc *Vspc
+}
+
+// dataHdlr is the telnet data handler
+func (h *handler) dataHdlr(w io.Writer, b []byte, tc *telnet.Conn) {
+	cvm, exists := h.vspc.cvmFromTelnetConn(tc)
+	if !exists {
+		log.Errorf("vspc datahandler cannot find the vm with the provided data")
+	}
+	cvm.remoteConn.Write(b)
+}
+
+// CmdHdlr is the telnet command handler
+func (h *handler) cmdHdlr(w io.Writer, b []byte, tc *telnet.Conn) {
+	switch {
+	case isKnownSuboptions(b):
+		log.Infof("vspc received KNOWN-SUBOPTIONS command")
+		h.handleKnownSuboptions(w, b)
+	case isDoProxy(b):
+		log.Infof("vspc received DO-PROXY command")
+		h.handleDoProxy(w, b)
+	case isVmotionBegin(b):
+		log.Infof("vspc received VMOTION-BEGIN command")
+		h.handleVmotionBegin(w, tc, b)
+	case isVmotionPeer(b):
+		log.Infof("vspc received VMOTION-PEER command")
+		h.handleVmotionPeer(w, b)
+	case isVMUUID(b):
+		log.Infof("vspc received VMUUID command")
+		h.handleVMUUID(w, tc, b)
+	case isVmotionComplete(b):
+		log.Infof("vspc received VMOTION-COMPLETE command")
+		h.handleVmotionComplete(w, tc, b)
+	case isVmotionAbort(b):
+		log.Infof("vspc received VMOTION-ABORT command")
+		h.handleVmotionAbort(w, tc, b)
```

Comment:
just wondering, should we log or panic in the default case?

Also maybe we could drop handle suffix from those functions as it stutters (h is already handler so h.DoProxy, h.VMUUID etc.)